synapse/storage/data_stores/main/events.py

```diff
@@ -1711,9 +1712,16 @@ def _purge_room_txn(self, txn, room_id):
 
         logger.info("[purge] done")
 
-    def purge_unreferenced_state_groups(self, room_id, state_groups_to_delete):
+    def purge_unreferenced_state_groups(
+        self, room_id: str, state_groups_to_delete: Set[int]
```

Comment:
does it have to be a Set, or is any iterable good enough?

Reply:
It needs to be an iterable that has a length, I don't know whether its better to just say iterable or say `List|Set` or something

Reply:
oh, that's a `Collection` I think.